lldb/tools/lldb-dap/Socket.h

```diff
@@ -0,0 +1,182 @@
+//===-- Socket.h ------------------------------------------------*- C++ -*-===//
```

Comment:
I'm not quite comfortable with this essentially reimplementing large parts of `source/Host/common/*Socket.cpp`.  @walter-erquinigo do you have any objections to using the implementation from the lldb `host` library. I see lldb-dap already depends on `lldbHost` on windows (and NetBSD?) -- this would make that dependency unconditional.

Reply:
Sounds good, I can re-work my patch to use `lldb/Hosts/*`

Reply:
I reworked this now to use the `lldb/Host/Socket.h` and associated types.

Reply:
Cool. Thanks.